src/graph/visitor/DeduceTypeVisitor.cpp

```diff
@@ -116,6 +116,10 @@ void DeduceTypeVisitor::visit(ConstantExpression *expr) {
 }
 
 void DeduceTypeVisitor::visit(UnaryExpression *expr) {
+  checkDepth();
+  if (!status_.ok()) {
+    return;
```

Comment:
should always recover depth when exit current function:
```c++
checkDepth();
SCOPED_EXIT {
  recoverDepth();
}
```

However I also don't think we should do this thing in this visitor.

Reply:
ok I will create a new visitor